### PR TITLE
API-12658 nod v2 form data

### PR DESCRIPTION
**Pull request by ElleeB**
Part 1 of ticket - Part 2 will update the structure

**Adds for V2:**
`form_data`
`form_data_spec`
v2 specific methods on nod model

**Changes:**
`veteran_representative` method name on nod model to `representative`
this change required a refactor to a v1 `form_data` method, but has no external impact

[Ticket](https://vajira.max.gov/browse/API-12658)

### CODE REVIEW

File: modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb [hunk 1]
```diff
@@ -164,18 +200,6 @@ def update_status!(status:, code: nil, detail: nil)
       email_handler.handle! if status == 'submitted' && email_identifier.present?
     end
 
-    def extension_request?
```
Comment (ElleeB) [inline modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb:167]: moved these methods up in the file

File: modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb [hunk 2]
```diff
@@ -91,18 +125,20 @@ def phone
...
-      veteran_contact_info['email'] || veteran_contact_info['emailAddressText']
+      # V2 and V1 access the email data via different keys ('email' vs 'emailAddressText')
+      veteran.email.presence || veteran_contact_info['emailAddressText']
     end
 
     def veteran_homeless?
       form_data&.dig('data', 'attributes', 'veteran', 'homeless')
     end
 
-    def veteran_representative
```
Comment (ElleeB) [inline modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb:101]: modified v1 form_data to use `representative`

**Review by cilestin — CHANGES_REQUESTED** [review]
Lookin' pretty good! Only see one thing in my quick scan of it. Will give it a deeper look when I have more time to spare.

File: modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb [hunk 3]
```diff
@@ -91,14 +125,15 @@ def phone
...
+    def representative
```
[the remark this answers is not in the document]
Comment (ElleeB) [reply modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb:136]: Ah! Nice. I'll update it.

**Review by cilestin — APPROVED** [review]
LGTM!

**Review by cilestin — APPROVED** [review]

**Pull request merged**